nav2_mppi_controller/CMakeLists.txt

```diff
@@ -1,22 +1,12 @@
 cmake_minimum_required(VERSION 3.5)
 project(nav2_mppi_controller)
 
-add_definitions(-DXTENSOR_ENABLE_XSIMD)
-add_definitions(-DXTENSOR_USE_XSIMD)
-
-set(XTENSOR_USE_TBB 0)
-set(XTENSOR_USE_OPENMP 0)
-set(XTENSOR_USE_XSIMD 1)
-
-# set(XTENSOR_DEFAULT_LAYOUT column_major)  # row_major, column_major
-# set(XTENSOR_DEFAULT_TRAVERSAL row_major)  # row_major, column_major
-
 find_package(ament_cmake REQUIRED)
-find_package(xsimd REQUIRED)
-find_package(xtensor REQUIRED)
+find_package(Eigen3 REQUIRED)
```

Comment:
Are the various `check_cxx_compiler_flag` useful for Eigen?

Reply:
-mfma is necessary for fast floating point fused multiply-add operations. [Answer by One of the Eigen Core maintainer](https://stackoverflow.com/questions/24296152/how-to-speed-up-this-c-program-with-eigen-library-against-matlab#:~:text=Finally%2C%20you%20might%20also%20consider%20using%20the%20devel%20branch%20and%20enable%20AVX%20(e.g.%2C%20%2Dmavx)%20and%20FMA%20if%20your%20CPU%20does%20support%20FMA%20(e.g.%2C%20%2Dmfma). He has also mentioned about enabling OpenMP for multi-threading.

Reply:
One interesting find, I tried removing the -fast-math flag and it speeds up the performance by a big margin.
For 2000 * 50 size:
xtensor: 11.6 ms avg.
Eigen with fast-math flag: 8.9 ms
Eigen without fast-math flag: 7.5 ms

Reply:
> And it seems there is no harm in keeping ISA flags(SSE/AVX).

There actually is, ARM based processors have inconsistent support for them or versions, so it makes release difficult. If they don't impact performance positively, we should remove them and removes an entire vector of potential problems (which we're currently running into in https://github.com/ros-navigation/navigation2/issues/4380 and a few other places in the past)

I'd be curious about OpenMP's changes in performance! We could make that a build option if it helps and those that want to use it can!

https://stackoverflow.com/questions/56547557/basic-ways-to-speed-up-a-simple-eigen-program there are some remarks here on AVX/fast-math, mfma. Might be worth doing a bit of research on Eigen-specific compiler optimizations, it looks like a rich vein and what was good for xtensor might not be right for Eigen. 

For fast-math, it might be worth testing on a couple of CPUs if you have them to make sure (if not, I can also test on my side, I have a few on my benchtop). What experiment are you running to get that performance change and/or does it include all the critics?

Reply:
> If they're important, lets keep them but I had hoped this would be a good source of removal of problems with Eigen

On my CPU only -mfma and -O3 impacted performance positively(and removal of fast-math also). There was no impact of AVX and SSE flags.

Reply:
> There actually is, ARM based processors have inconsistent support for them or versions, so it makes release difficult.

Oh, I wasn't aware of this. Once merge conflicts and build errors are resolved, then we can test on multiple CPUs with different flags.

Reply:
Hey @SteveMacenski, after fixing all the tests and system tests, these are the timings that I'm getting on my machine. I'm using optimizer_benchmark.cpp to compare the timings.

batch_size = 2000;
time_steps = 56;
path_points = 50;
iteration_count = 2;
lookahead_distance = 20;
critics = {{"ConstraintCritic"}, {"CostCritic"}, {"GoalCritic"},
    {"GoalAngleCritic"}, {"PathAlignCritic"}, {"PathFollowCritic"}, {"PathAngleCritic"}, {"PreferForwardCritic"}};
![image](https://github.com/user-attachments/assets/ca691780-e874-462e-843e-9c36f95d776c)
The result looks good. Can you try the PR on your machine and check if similar gains are reproducible on your machine also?

Reply:
Wow. So you're seeing a ~47% increase in performance? That's incredible. What CPU did you test this against?

Reply:
> What CPU did you test this against?

It is Intel i7-8750H 12 cores.